Detectors/ZDC/simulation/src/Digits2Raw.cxx

```diff
@@ -0,0 +1,704 @@
+    UShort_t bc = mBCD.ir.bc;
+    UInt_t orbit = mBCD.ir.orbit;
+    // Look for channel ID in digits and store channel (just one copy in output)
+    // This is a limitation of software but we are not supposed to acquire the
+    // same signal twice anyway
+    for (Int_t im = 0; im < NModules; im++) {
+      for (UInt_t ic = 0; ic < NChPerModule; ic++) {
+        if (mModuleConfig->modules[im].channelID[ic] == chd.id &&
+            mModuleConfig->modules[im].readChannel[ic]) {
+          Int_t is = 0;
+          mZDC.data[im][ic].f.s00 = chd.data[is];
+          is++;
+          mZDC.data[im][ic].f.s01 = chd.data[is];
+          is++;
+          mZDC.data[im][ic].f.s02 = chd.data[is];
+          is++;
+          mZDC.data[im][ic].f.s03 = chd.data[is];
+          is++;
+          mZDC.data[im][ic].f.s04 = chd.data[is];
+          is++;
+          mZDC.data[im][ic].f.s05 = chd.data[is];
+          is++;
+          mZDC.data[im][ic].f.s06 = chd.data[is];
+          is++;
+          mZDC.data[im][ic].f.s07 = chd.data[is];
+          is++;
+          mZDC.data[im][ic].f.s08 = chd.data[is];
+          is++;
+          mZDC.data[im][ic].f.s09 = chd.data[is];
+          is++;
+          mZDC.data[im][ic].f.s10 = chd.data[is];
+          is++;
+          mZDC.data[im][ic].f.s11 = chd.data[is];
+          break;
+        }
+      }
+    }
+  }
+}
+
+//______________________________________________________________________________
+void Digits2Raw::writeDigits()
+{
+  constexpr static int data_size = sizeof(uint32_t) * NWPerGBTW;
+  // Local interaction record (true and empty bunches)
+  o2::InteractionRecord ir(mZDC.data[0][0].f.bc, mZDC.data[0][0].f.orbit);
+  for (UInt_t im = 0; im < o2::zdc::NModules; im++) {
+    // Check if module has been filled with data
+    // N.B. All channels are initialized if module is supposed to be readout
+    // Trigger bits are the same for all the channels connected to a module
+    bool TM = mZDC.data[im][0].f.Auto_m;
+    bool T0 = mZDC.data[im][0].f.Auto_0;
+    bool T1 = mZDC.data[im][0].f.Auto_1;
+    bool T2 = mZDC.data[im][0].f.Auto_2;
+    bool T3 = mZDC.data[im][0].f.Auto_3;
+    bool A0 = mZDC.data[im][0].f.Alice_0;
+    bool A1 = mZDC.data[im][0].f.Alice_1;
+    bool A2 = mZDC.data[im][0].f.Alice_2;
+    bool A3 = mZDC.data[im][0].f.Alice_3;
+    bool tcond_continuous = T0 || T1;
+    bool tcond_triggered = A0 || A1 || (A2 && (T0 || TM)) || (A3 && T0);
+    bool tcond_last = mZDC.data[im][0].f.bc == 3563;
+    // Condition to write GBT data
+    if (tcond_triggered || (mIsContinuous && tcond_continuous) || (mZDC.data[im][0].f.bc == 3563)) {
+      for (UInt_t ic = 0; ic < o2::zdc::NChPerModule; ic++) {
+        if (mModuleConfig->modules[im].readChannel[ic]) {
+          for (Int_t iw = 0; iw < o2::zdc::NWPerBc; iw++) {
+            gsl::span<char> payload{reinterpret_cast<char*>(&mZDC.data[im][ic].w[iw][0]), data_size};
+            mWriter.addData(mFeeID, mCruID, mLinkID, mEndPointID, ir, payload);
```

Comment:
@cortesep In the test I see that only 1 link with ``feeID=0xf`` gets some data and in the code I don't see where do you set this ``mFeeID``: it remains on 0xf because this was the last registered link.
I was testing as:
````
2-sim -m ZDC PIPE -n 5 -g pythia8hi  --configKeyValues "Diamond.width[2]=6.; SimCutParams.maxRTracking=100;"
o2-sim-digitizer-workflow  --interactionRate 10000
o2-zdc-digi2raw -o raw/ZDC --file-per-link 
o2-raw-file-check --input-conf raw/ZDC/ZDCraw.cfg -v2 | grep Offs | grep -v 'OffsNext:64'
[INFO] EP:0 CRU:0x0000 Link:0   FEEID:0x000f SrcID:ZDC[15] Packet:4   MemSize:4960  OffsNext:4960   prio.:0 HS:64 HV:6
[INFO] EP:0 CRU:0x0000 Link:0   FEEID:0x000f SrcID:ZDC[15] Packet:6   MemSize:1312  OffsNext:1312   prio.:0 HS:64 HV:6
[INFO] EP:0 CRU:0x0000 Link:0   FEEID:0x000f SrcID:ZDC[15] Packet:8   MemSize:2464  OffsNext:2464   prio.:0 HS:64 HV:6
[INFO] EP:0 CRU:0x0000 Link:0   FEEID:0x000f SrcID:ZDC[15] Packet:10  MemSize:3136  OffsNext:3136   prio.:0 HS:64 HV:6
[INFO] EP:0 CRU:0x0000 Link:0   FEEID:0x000f SrcID:ZDC[15] Packet:12  MemSize:2560  OffsNext:2560   prio.:0 HS:64 HV:6

````